### ISSUE TITLE
[QUESTION]  Participation in official Addons Store Beta Test

### ISSUE THREAD

**Issue by patrickgold** [itsmartashub/floristyle#11]
### Prerequisites (Required)

- [X] I [searched for any existing report](https://github.com/itsmartashub/floristyle/issues?q=is%3Aissue) about this question to avoid opening a duplicate.
- [X] This issue contains only one question.
- [X] The title of this issue accurately describes the question.

### What would you like to share? (Required)

Hi,
my team and I are currently finishing up the preparations for the beta test launch of the official FlorisBoard Addons Store, and I wanted to ask if you want your Floristyle themes to be part of the initial launch?

**Comment by itsmartashub** [comment]
Hey Patrick, thanks so much for reaching out and for considering floristyle themes for the initial launch of the FlorisBoard Addons Store. It would be my pleasure. I wpuld be thrilled to have floristyle themes included in the beta test launch.

Please let me know if there are any specific steps or requirements I need to follow to make this happen.

PS. Just I probably wont have access to the internet for the next two weeks starting tomorrow 😕

**Comment by patrickgold** [comment]
Great to hear you want to be part of the initial release! We probably won't manage to open up the publication process today tough, so that might be a bit if a problem for the initial launch. I will report later on how things go.

**Comment by patrickgold** [comment]
In general though the changed requirements are that the extension ID starts with
```io.github.<usernam>.****```
or you use a custom domain verified by a TXT record as a prefix. Other than that your extension should afaik be ready for the addons store

**Comment by itsmartashub** [comment]
Thank you for keeping me updated on the progress. Regarding the extension ID format, I will ensure that it follows the specified structure.

I'm just wondering what add-on creators should know about the publication process at this stage.
Does this mean that the FlorisBoard AddOns Store is basically ready for the production and we'll be able to register soon? Could you please explain the process in more detail?
Like, will we first register and then submit a "request" for our extension as creators, providing the necessary information, images, etc.? Or will the extension data be automatically pulled from the GitHub repo/source code? Or is there another process in mind?

I'm just checking in to see if there's anything I can do tonight to get ready, like necessary files, prep images or using the ones on GitHub would be fine, etc. That way, I'll have some things I might need on my phone since I won't have access to a PC and the internet strength would be very poor.

**Comment by patrickgold** [comment]
Yes, the addons store is near a beta launch of the addons store. I will publish a detailed wiki entry later on, but basically:
1. You register on the addons store using your GitHub account
2. If wanted, you register your custom domain
3. You create a new publishing project, add the meta data, image (1) & screenshots (0-many)
4. Then the project is created, and you can add releases
5. Upon creating a release you upload the flex file which needs the correct ID and version set
6. Then the release must for the time being be manually reviewed by us and once verified, published

**Comment by patrickgold** [comment]
I just had a look at the v3.0.1 flex file and the meta data looks ready for publishing to me

**Comment by itsmartashub** [comment]
Thanks for the details about the process. I really appreciate your efforts and time in putting together a wiki entry to guide us through the steps.  
  
Can't wait to see the FAS (FlorisBoard AddOns Store) come to life and thank you for the opportunity to contribute FloriStyle themes to the community in this early stage.
  
PS thanks for taking the time to review the `v3.0.1` metadata

**Comment by itsmartashub** [comment]
Hey, @patrickgold, I have a query concerning the registration process. 
As someone who is apprehensive about registering with my primary email address anywhere, especially through providers, and who prefers utilizing forwarding mail services, I am curious about the rationale behind the exclusive registration via GitHub. 

- Is this approach intended to facilitate user verification and deter bot accounts?
- Are all users required to register in this manner, or is it solely intended for extension creators to streamline the connection to their source code? 
- What if creators opt for alternative platforms like GitLab or Codeberg for their source code management? Does FAS necessitate that all creators utilize GitHub for their source code? 
- Would non-creator users, such as those who wish to register, leave reviews, save fav, etc., also need to create a GitHub account?

I'm sorry for "overquestioning"

**Comment by patrickgold** [comment]
> Is this approach intended to facilitate user verification and deter bot accounts?

Yes, one reason why we only allow GitHub accounts for now is to deter spammers at least a bit.

> Are all users required to register in this manner, or is it solely intended for extension creators to streamline the connection to their source code?

All users that intent to perform actions requiring a sign in, be it users that want to review extensions or extension creators must sign up via GitHub.

> What if creators opt for alternative platforms like GitLab or Codeberg for their source code management? Does FAS necessitate that all creators utilize GitHub for their source code?

You can host your source code and manage it wherever you want, the GitHub account requirement only concerns the account itself and not the created publishing projects. This will also include the publishing API later on, you can trigger releases via the API from anywhere.

> Would non-creator users, such as those who wish to register, leave reviews, save fav, etc., also need to create a GitHub account?

Yes, see above.

Hope that clears some of your questions/concerns, else don't hesitate to ask!

**Comment by itsmartashub** [comment]
Thanks for this Q&A session and taking the time to explain things further about user verification and requirements

**Comment by patrickgold** [comment]
Took us longer than expected but the Addons Store is finally ready! It can be accessed at https://beta.addons.florisboard.org

[This wiki entry](https://github.com/florisboard/florisboard/wiki/How-to-publish-on-FlorisBoard-Addons) is a good guide for publishing, if you need assistance don't hesitate to ask me!

**Comment by itsmartashub** [comment]
Thanks for the update. Great to hear the Addons Store is up and running. Well done! 🥳

I understand that this is still in the very early stages, but is there a plan to move the login/register button to a more visible spot on the website?

Also, I created a theme extension project, but some things remain unclear after the editing phase. 
Initially, I was able to edit the project (for example, reupload different images). However, after clicking on `Publish`, when I return to my profile, the images for that project are not refreshed/reuploaded, and it is no longer possible to re-enter the editing process.
Could this be due to the manual review process? Will the project become editable again after the review is completed?

**Comment by patrickgold** [comment]
Congrats for publishing, looks good!

> I understand that this is still in the very early stages, but is there a plan to move the login/register button to a more visible spot on the website?

Not directly, but if we receive more reports of this we might consider moving the login button out of the dropdown.

> However, after clicking on Publish, when I return to my profile, the images for that project are not refreshed/reuploaded, and it is no longer possible to re-enter the editing process.

This is because if you click "Publish" your project's status is under review, meaning we need to manually review the queued changes. Once approved (or theoretically rejected) you can edit the metadata again as before. We are aware that the UX of this is sup-optimal in the current state, however we have yet to figure out how o make this process more smooth.

**Comment by itsmartashub** [comment]
> This is because if you click "Publish" your project's status is under review, meaning we need to manually review the queued changes. Once approved (or theoretically rejected) you can edit the metadata again as before. We are aware that the UX of this is sup-optimal in the current state, however we have yet to figure out how o make this process more smooth.

Yes, something like how the web extension dev dashboards function and makes perfect sense. 
Perhaps, for now, it wouldn't be a bad idea to disable or hide the `Edit` button on the project page while the extension is under review.
Also, when in edit mode, a `Submit for Review` or at least `Done` label might provide clearer guidance to the user regarding the button's function, rather than the `Edit` label when already in the edit layout.

### ADDITIONAL
I understand this is an early stage and not the main focus at the moment, but it might be beneficial in the future to add a full "Preview" layout for extension creators (and a huge thanks and shoutout to the developers for implementing description in **markdown** format and its preview! 🥳). This would help avoid unnecessary edit spam reviews, making the process easier for both reviewers and creators, and would also save time for both parties.
For instance, in my specific case, it would have avoided several edits and review processes simply because I was unsure how the uploaded images would fit the given interface, such as image aspect ratios, or what happening to screenshots, etc

P.S. Just to confirm, are the project main image and additional screenshots also editable from the backend? On the frontend, images are changed while in edit state, but after editing and the review process is completed, no changes are reflected (I've already spammed reviews twice only because of this. Sorry for that dear reviewers 🥲). It seems I might be doing something wrong, given that the second review process has passed without added/edited images.

But to avoid spamming you further here, perhaps it’s better to discuss this in the Matrix group or a Florisboard Add On GitHub discussion.

**Comment by patrickgold** [comment]
I like your idea of a full preview layout to avoid unnecessary "spam" reviews, I will pass that on to the team!

Regarding the screenshots, we are currently investigating what happened here, for now I suspect a file bug somewhere, will notify you once we figured out the reason.

**Comment by patrickgold** [comment]
Can you try again uploading the screenshots and sending them from review?

**Comment by 4H1R** [comment]
What's up first of all thank you for your contribution to addons stores

The screenshots seems to be working but it might be random bug that we cannot reproduce 

If that's the case do we have the permission to upload screenshots to your project for testing purposes?

**Comment by itsmartashub** [comment]
> Can you try again uploading the screenshots and sending them from review?

No luck again, I guess, but thank you for keeping me informed.

Hey @4H1R, thank you for giving your time and expertise to this project. Congratulations on the official production release

As for the screenshot editing problem, I still haven't been able to solve it. The images are in `png` format. 
- What I've been doing, including now, is accidentally uploading an image I didn't want in the sc and then removing it. 
- This triggers a cascade of messages: `Something went wrong. Please try again later`, the image disappears from the front-end and then being automatically re-uploaded.
- I then uploaded another image and clicked `Edit` again (in the sense of submitting for review), but then two notifications appeared at the same time, the precise content of which I can't remember, and that part wasn't recorded. One notif was green, indicating a successful submission for the review I guess, while the red notif said something like that images couldnt be submitted during the review process, which confused me further.

Next time I will try to capture everything, and without removing the uploaded image.


https://github.com/user-attachments/assets/09e44bbd-b01f-47bd-8337-6cc1817eb461

**Comment by lm41** [comment]
> Sure. Let's confuse the Florisboarders so please feel free to upload any cat typing on a keyboard screenshots you like. At least they might be beneficial for reviews ("kittying" ofc)

I think what ali meant is that you send one of the admins (patrick, ali or me) the screenshots and we will add them. You can also upload them here in the issue and I can make sure they end up where they should be.

**Comment by itsmartashub** [comment]
Yes, that’s how I understood it too and I’ve already sent them. But thank you for clarifying the situation in case I was serious.

> Edit: This is a bit of a long shot, but could it be that the whitespace in the screenhot names is causing a problem?

Yes, noticed that only later and sent the images to the admin with simple img names without spacing and other spec characters. I hope that was the issue all the time, because if so, then preventing this (image names, extra characters, spacing, etc) shouldn’t be too complicated to handle from the developers' side I guess.

**Comment by 4H1R** [comment]
I see thank you for the footage and the details you've provided 
We've investigate and see what's up

We'll update you here

**Comment by lm41** [comment]
> Yes, that’s how I understood it too and I’ve already sent them. But thank you for clarifying the situation in case I was serious.

I've seen everything on github, so I wanted to make sure even if I was 90% sure it was a joke

**Comment by itsmartashub** [comment]
@lm41 Oh no, I can only imagine. I must be missing out on some top-tier github drama? 👀😂 someone pass the popcorn pls🍿

But for real, thanks for letting me be a tiny cog in this awesome keyboard machin 🫂 I appreciate you all 💟
